### PR TITLE
MINOR: Rename baseTimestamp to firstTimestamp to clarify usage

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

**Comment by hachikuji** [comment]
cc @ijuma @apurvam

File: clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java [hunk 1]
```diff
@@ -140,8 +140,8 @@ public void ensureValid() {
      *
      * @return The base timestamp
      */
-    public long baseTimestamp() {
-        return buffer.getLong(BASE_TIMESTAMP_OFFSET);
+    public long firstTimestamp() {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java]: return in javadoc needs to be updated

**Review by ijuma — COMMENTED** [review]
This makes sense to me, what do you think @junrao?

File: clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java (hunk 1 above)
Comment (hachikuji) [reply clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java]: Ack. I will also update the javadoc in this class to explain the difference in usage between `baseOffset` and `firstTimestamp`

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5782/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5768/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by junrao** [comment]
@ijuma : Yes, this makes sense to me.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5789/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5775/
Test PASSed (JDK 8 and Scala 2.12).

File: clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java [hunk 2]
```diff
@@ -64,12 +64,23 @@
...
+ * last sequence numbers of the incoming batch match the last from that producer.
+ *
+ * Note that if all of the records in a batch are removed during compaction, the broker may still retain an empty
+ * batch header in order to preserve the producer sequence information as described above. These empty batches
+ * are retained only until either a new sequence number is written by the corresponding producer or the producerId
+ * is expired from lack of activity.
+ *
+ * There is no similar need to preserve the timestamp from the original batch after compaction. The FirstTimestamp
+ * field therefore always reflects the timestamp of the first record in the batch. If the batch is empty, the
+ * FirstTimestamp will be set to -1 (NO_TIMESTAMP).
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java:83]: Should we mention MaxTimestamp here?

**Review by ijuma — APPROVED** [review]
LGTM, just one comment.

File: clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java (hunk 2 above)
Comment (hachikuji) [reply clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java:83]: Couldn't hurt.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5827/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5813/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by ijuma** [comment]
Merged to trunk and 0.11.0.

**Pull request closed without merging**